### DISCUSSION TITLE
How can I access the gRPC Mocks described in the documentation?

### DISCUSSION THREAD

**Discussion by suic86** [microcks/microcks#723] [Q&A] [answered]
I'm using the the [grpc example](https://microcks.io/documentation/using/grpc/) from the documentation. In microcks web console it looks fine (see below).

When I try to access the it on localhost:9090 via Postman it doesn't work. I'm using to the docker compose installation. How can I access the gRPC mock?

Thanks!

 
gRPC Service in microcks console:

![image](https://user-images.githubusercontent.com/11985251/207714261-6f7daa14-279a-4745-af7e-9fedcbefd682.png)

Call from Postman using the proto file:

![image](https://user-images.githubusercontent.com/11985251/207714997-f29d7eaf-8f09-47d4-8c5d-30bf78875889.png)

Trying to use reflection in Postman:

![image](https://user-images.githubusercontent.com/11985251/207715761-c22a00a8-35ed-4830-9b18-adbb6c134b0f.png)

**Comment by lbroudoux** [comment, marked as answer]
You've got the correct URL but I think you got an exception because Postman GRPC support might be based on server reflection protocol that is not mandatory and not currently implemented into Microcks ... You may check everything is working correctly with another GRPC client like Bloom RPC.

I'll check to see what implementing server reflection actually means.

**Reply by suic86** [reply]
Thanks for the reply.

I'm getting the same with BloomRPC:

![image](https://user-images.githubusercontent.com/11985251/209391179-2f2f6e38-162e-4955-a6f7-f4a398fb5158.png)

I'm running the docker compose inside WSL and accessing the endpoint from W10. I'm not sure if this can affect/cause the issue. However, the REST services I've tested work well with this setup.

**Reply by lbroudoux** [reply]
Hi!
I've looked back at the previously posted screenshot and I think you've got this error on Bloom RPC because there's actually no dispatcher configured for your mock. Look my definition below that is slightly different.

<img width="1442" alt="Capture d’écran 2022-12-26 à 11 22 58" src="https://user-images.githubusercontent.com/1538635/209538061-e0bdd496-dab4-45a8-ba90-e9691c5918e3.png">

To change the dispatcher, you have 2 different options:
* Edit the operation properties from the UI and setting the dispatcher manually (see [here](https://microcks.io/documentation/using/advanced/dispatching/#dispatching-rules-override)),
* Uploading a specific [APIMetadata](https://microcks.io/documentation/using/advanced/metadata/#apimetadata-format) file as a second artifact for your GRPC service definition. You can find such a file for `Hello Service` [here](https://github.com/microcks/microcks/raw/master/samples/HelloService.metadata.yml).

Let me know if that helps,

**Reply by suic86** [reply]
Hi!

Thanks! This was the problem. I had no dispatching rules configured. Now it works with BloomRPC and in Postman (see below)

![image](https://user-images.githubusercontent.com/11985251/210212448-d8bece99-25d9-4c85-a6ce-bef610a18add.png)
